### PR TITLE
[tune] experiment_analysis split to Analysis

### PR DESCRIPTION
**Pull request by richardliaw**
## What do these changes do?
Users have mentioned that the analysis interface is confusing. This
changes the ExperimentAnalysis interface to take in a json file path.

## Linter

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by sotte** [comment]
The change in `ExperimentAnalysis` is straight forward and makes sense!

Using the CLI of tune to list experiences is still exhibits the confusing behaviour of selecting the most recent experiment. As in #5090 I would expect that I either specify a concrete experiment (i.e. the json file), or I get all the experiments and trials.

@richardliaw are you planning to add the generic `Analysis` class that you mentioned in #5090 in the PR as well, or in a sep. PR?

**Comment by richardliaw** [comment]
> Using the CLI of tune to list experiences is still exhibits the confusing behaviour of selecting the most recent experiment. As in #5090 I would expect that I either specify a concrete experiment (i.e. the json file), or I get all the experiments and trials.

OK Let me address that in this PR. 

> @richardliaw are you planning to add the generic Analysis class that you mentioned in #5090 in the PR as well, or in a sep. PR?

I think I'll do this part in a separate PR. 

Thanks!

**Comment by richardliaw** [comment]
@sotte I've made a couple changes, but still have quite a bit to do to make this user-friendly (specifically for CLI). I think the Python API is ready to go - let me know if you give it a test run.

If you have time, do you have any thoughts on what should be displayed in the default columns for the CLI?

**Comment by sotte** [comment]
I like this much better than before. Giving it a test run...and will get back to you.

File: python/ray/tune/analysis/experiment_analysis.py [hunk 1]
```diff
@@ -34,103 +34,101 @@ def unnest_checkpoints(checkpoints):
...
-        experiment_path (str): Path to where experiment is located.
-            Corresponds to Experiment.local_dir/Experiment.name
-
-    Example:
-        >>> tune.run(my_trainable, name="my_exp", local_dir="~/tune_results")
-        >>> analysis = ExperimentAnalysis(
-        >>>     experiment_path="~/tune_results/my_exp")
-    """
+class Analysis(object):
+    """Analyze results from directory of experiments."""
```
Comment (sotte) [inline python/ray/tune/analysis/experiment_analysis.py]: Maybe: "Analyze  *all* results from a directory of experiments."

File: python/ray/tune/analysis/experiment_analysis.py [hunk 2]
```diff
@@ -139,18 +137,48 @@ def get_best_logdir(self, metric, mode="max"):
...
             return df.iloc[df[metric].idxmin()].logdir
 
-    def get_best_info(self, metric, mode="max", flatten=True):
-        """Retrieve the best trial based on the experiment metric.
+
+class ExperimentAnalysis(Analysis):
+    """Analyze results from a Tune experiment.
+
+    Parameters:
+        experiment_path (str): Path to where experiment is located.
```
Comment (sotte) [inline python/ray/tune/analysis/experiment_analysis.py]: Maybe: "Path to a json file representing an experiment state"

At least we should point out that is has to be the json file, not just a generic path. We could also rename `experiment_path` accordingly. The example below does not use the json file.

**Review by sotte — COMMENTED** [review]
In general I wonder if there are better names than `Analysis` and `ExperimentAnalysis`. It's not obvious what the difference between the two is. But I also don't know a better name.

Regarding the CLI I have to say, that I don't really have too much of an opinion (except that the behaviour should be not surprising).

**Comment by sotte** [comment]
Also in the current version:
```bash
$ tune lsx some_path
Traceback (most recent call last):
...
AttributeError: 'Analysis' object has no attribute 'stats'
```

**Comment by richardliaw** [comment]
@sotte fixed - take a test run?

**Review by ericl — APPROVED** [review]

**Comment by richardliaw** [comment]
jenkins retest this please

**Pull request merged**